### PR TITLE
Allow using a period in a CLI product name

### PR DESCRIPTION
**Pull request by serejahh**
### Short description 📝

Here's one more fix for macOS developers 😅

A macOS app can use privileged helper tools that are run under admin.
The tools are stored in `/Library/PrivilegedHelperTools` and have a strict format: `<bundle_id>.name`. So a period is a legal symbol for such tools

Here's an example:
image

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [ ] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 1]
```diff
@@ -73,14 +73,20 @@ class TargetLinter: TargetLinting {
     private func lintProductName(target: Target) -> [LintingIssue] {
         var allowed = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_")
 
-        if target.product == .app {
+        if target.product == .app || target.product == .commandLineTool {
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:76]: nit: I would define and assing a `let allowsDot = target.product == .app || target.product == .commandLineTool` variable here, and reuse it below

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 2]
```diff
@@ -73,14 +73,20 @@ class TargetLinter: TargetLinting {
...
-                "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9), and underscore (_) characters."
+            let reason: String
+            switch target.product {
+            case .app, .commandLineTool:
+                reason =
+                    "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9), period (.), and underscore (_) characters."
+            default:
+                reason =
+                    "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9), and underscore (_) characters."
+            }
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:89]:
```suggestion
            let reason = "Invalid product name '\(target.productName)'. This string must contain only alphanumeric (A-Z,a-z,0-9)\(allowsDot ? ", period (.)" : ""), and underscore (_) characters."
```

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistGenerator/Linter/TargetLinter.swift (hunk 1 above)
Comment (serejahh) [reply Sources/TuistGenerator/Linter/TargetLinter.swift:76]: Thanks for the comments!

**Comment by serejahh** [comment]
Don't you mind adding me to the contributors list? 🙏

**Comment by danyf90** [comment]
@allcpntributors add @serejahh for code

**Comment by danyf90** [comment]
@allcontributors add @serejahh for code

**Pull request merged**